lib/cli-engine.js

```diff
@@ -391,12 +392,14 @@ function getCacheFile(cacheFile, cwd) {
  */
 function CLIEngine(options) {
 
+    options = (options instanceof Object) ? assign(options) : {};
+    options.cwd = options.cwd || process.cwd();
+
     /**
      * Stored options for this instance
      * @type {Object}
      */
-    this.options = assign(Object.create(defaultOptions), options || {});
-
+    this.options = assign(defaultOptions, options);
```

Comment:
`Object.create(defaultOptions)` was always empty. Does the following change address your comment?
```
this.options = assign(defaultOptions, options || {});
```

Under which circumstances do you expect undefined values? The examples from [object-assign](https://github.com/sindresorhus/object-assign/blob/master/readme.md) show that an undefined value would just be ignored.